iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/TieLineUtil.java

```diff
@@ -19,9 +29,114 @@
+
     private TieLineUtil() {
     }
 
+    public static String buildMergedId(String id1, String id2) {
+        if (id1.compareTo(id2) < 0) {
+            return id1 + " + " + id2;
+        }
+        if (id1.compareTo(id2) > 0) {
+            return id2 + " + " + id1;
+        }
+        return id1;
+    }
+
+    public static String buildMergedName(String id1, String id2, String name1, String name2) {
+        if (name1 == null) {
+            return name2;
+        }
+        if (name2 == null) {
+            return name1;
+        }
+        if (name1.compareTo(name2) == 0) {
+            return name1;
+        }
+        if (id1.compareTo(id2) < 0) {
+            return name1 + " + " + name2;
+        }
+        if (id1.compareTo(id2) > 0) {
+            return name2 + " + " + name1;
+        }
+        if (name1.compareTo(name2) < 0) {
+            return name1 + " + " + name2;
+        }
+        return name2 + " + " + name1;
+    }
+
+    public static void mergeProperties(DanglingLine dl1, DanglingLine dl2, Properties properties) {
+        Set<String> dl1Properties = dl1.getPropertyNames();
+        Set<String> dl2Properties = dl2.getPropertyNames();
+        Set<String> commonProperties = Sets.intersection(dl1Properties, dl2Properties);
+        Sets.difference(dl1Properties, commonProperties).forEach(prop -> properties.setProperty(prop, dl1.getProperty(prop)));
+        Sets.difference(dl2Properties, commonProperties).forEach(prop -> properties.setProperty(prop, dl2.getProperty(prop)));
+        commonProperties.forEach(prop -> {
+            if (dl1.getProperty(prop).equals(dl2.getProperty(prop))) {
+                properties.setProperty(prop, dl1.getProperty(prop));
+            } else if (dl1.getProperty(prop).isEmpty()) {
+                LOGGER.debug("Inconsistencies of property '{}' between both sides of merged line. Side 1 is empty, keeping side 2 value '{}'", prop, dl2.getProperty(prop));
+                properties.setProperty(prop, dl2.getProperty(prop));
+            } else if (dl2.getProperty(prop).isEmpty()) {
+                LOGGER.debug("Inconsistencies of property '{}' between both sides of merged line. Side 2 is empty, keeping side 1 value '{}'", prop, dl1.getProperty(prop));
+                properties.setProperty(prop, dl1.getProperty(prop));
+            } else {
+                LOGGER.debug("Inconsistencies of property '{}' between both sides of merged line. '{}' on side 1 and '{}' on side 2. Removing the property of merged line", prop, dl1.getProperty(prop), dl2.getProperty(prop));
+            }
+        });
+        dl1Properties.forEach(prop -> properties.setProperty(prop + "_1", dl1.getProperty(prop)));
+        dl2Properties.forEach(prop -> properties.setProperty(prop + "_2", dl2.getProperty(prop)));
+    }
+
+    public static void checkAssociatedDanglingLines(DanglingLine dl2, Function<String, DanglingLine> getDanglingLineById,
```

Comment:
I was a bit lost at first with the chosen names, I think it would help if we understand from them that the first argument is from the network currently added, and that the 2nd and 3rd are from the base network

Besides, couldn't we pass directly a `DanglingLine` instead of `getDanglingLineById` function?

Reply:
We do not only check but also merge, should we rename to something with merge? Or maybe we should split it in two parts, `Optional<DanglingLine> findAssociatedDanglingLine(DanglingLine dl2, Function<String, DanglingLine> getDanglingLineById, Function<String, List<DanglingLine>> getDanglingLinesByXnodeCode)`, and then merge it if present?